docs/javascripts/stats_calculator.js

```diff
@@ -1,130 +1,96 @@
-		} else if (className == "rogue") {
-			maxStats = maxRogue;
-		} else if (className == "warrior") {
-			maxStats = maxWarrior;
-		} else if (className == "wizard") {
-			maxStats = maxWizard;
-		}
-
-		targetStrBox.value = maxStats[STR];
-		targetIntBox.value = maxStats[INT];
-		targetWisBox.value = maxStats[WIS];
-		targetConBox.value = maxStats[CON];
-		targetDexBox.value = maxStats[DEX];
-	}
-
-	calculateButton.onclick = function() {
-
-		var hp = parseInt(HPBox.value);
-		var transcended = transcendedCheckbox.checked;
-
-		var currentStr = parseInt(currentStrBox.value);
-		var currentInt = parseInt(currentIntBox.value);
-		var currentWis = parseInt(currentWisBox.value);
-		var currentCon = parseInt(currentConBox.value);
-		var currentDex = parseInt(currentDexBox.value);
-
-		var targetStr = parseInt(targetStrBox.value);
-		var targetInt = parseInt(targetIntBox.value);
-		var targetWis = parseInt(targetWisBox.value);
-		var targetCon = parseInt(targetConBox.value);
-		var targetDex = parseInt(targetDexBox.value);
-
-		var current = [currentStr, currentInt, currentWis, currentCon, currentDex];
-		var target = [targetStr, targetInt, targetWis, targetCon, targetDex];
-
-		if (isNaN(hp) || hp <= 0) { return; }
-
-		total = 0;
-
-		for (let i = 0; i < current.length; i++)
-		{
-			if (isNaN(current[i]) || isNaN(target[i])) { continue; }
-			if (current[i] >= target[i] || current[i] <= 0) { continue; }
-
-			// Pre-Max Part:
-			if (current[i] < maxStats[i])
-			{
-				// Sum of Consequtive Numbers Formula
-				n = Math.min(target[i], maxStats[i]) - current[i];
-				xp = n * (current[i] + (Math.min(target[i], maxStats[i]) - 1)) / 2; // Summation Part
-
-				if (transcended) {
-					xp = xp * hp * POST_TRANSCENDED; // Multiplicative Factor
-				}
-				else {
-					xp = xp * hp * PRE_TRANSCENDED; // Multiplicative Factor
-				}
-			
-				total += xp;
-			}
-
-			// Post-Max Part:
-			// Sum of Consequtive Numbers Formula
-			n = target[i] - Math.max(current[i], maxStats[i]);
-                        xp = n * (Math.max(current[i], maxStats[i]) + (target[i] - 1)) / 2; // Summation Part
-
-			xp = xp * POST_MAX;
-			total += xp;
-
-		}
-
-		XPBox.value = total.toLocaleString("en-US");
-
-	}
-
-}
-
-if (document.getElementById("stats-calculator") != null) {
-	statsCalculator()
+  /**
+   * Offset backwards by 1 on each value, should be ready like this:
+   * Sum(1) = [0,1) = 0
+   * Sum(10) = [0,10) = 0+1+2+3+4+5+6+7+8+9
+   * Sum(4,6) = [4,6) = 4+5
+   * @param {Number} n 
+   * @param {Number} m 
+   * @returns 
+   */
+  function Sum(n, m) {
+    if (!m) {
+      m = n;
+      n = 0;
+    }
+    return 0.5 * (m - n) * (m + n - 1);
+  }
+
+  function PreStatCost(HP, MULT, CURRENT, TARGET) {
+    const XP_MULTIPLIER = HP * MULT;
+    return XP_MULTIPLIER * Sum(CURRENT, TARGET);
+  }
+
+  function PostStatCost(CURRENT, TARGET) {
+    return POST_MAX * Sum(CURRENT, TARGET);
+  }
+
+  function StatCost(HP, MULT, MAX, current, target) {
```

Comment:
Function names should be camelCase and ideally start with a verb (perhaps "calculate"):

https://www.robinwieruch.de/javascript-naming-conventions